bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAuthInfoImpl.java

```diff
@@ -41,13 +40,13 @@
     private final byte[] privateKey;
     private final String pin;
 
-    public HomekitAuthInfoImpl(StorageService storageService, String pin) throws InvalidAlgorithmParameterException {
-        storage = storageService.getStorage("homekit");
-        initializeStorage();
+    public HomekitAuthInfoImpl(final Storage storage, final String pin) throws InvalidAlgorithmParameterException {
+        this.storage = storage;
         this.pin = pin;
-        mac = storage.get("mac");
-        salt = new BigInteger(storage.get("salt"));
-        privateKey = Base64.getDecoder().decode(storage.get("privateKey"));
+        initializeStorage();
+        mac = this.storage.get("mac");
+        salt = new BigInteger(this.storage.get("salt"));
+        privateKey = Base64.getDecoder().decode(this.storage.get("privateKey"));
```

Comment:
Seems wasteful to make multiple calls to `storage.get` during initialization. You have no idea what kind of storage is used nor how performant that storage is. Please try to minimize interactions with storage as much as possible.

Reply:
i dont think we can really delay the gets and make kind of lazy loading. homekit need mac, salt and private key to start the server. 
but i could reduce number of gets by moving them to initializeStorage. please take a look.